woodwork/type_sys/inference_functions.py

```diff
@@ -17,26 +17,22 @@ def get_inference_sample(series: pd.Series) -> pd.Series:
 
 
 def categorical_func(series):
-    categorical_threshold = ww.config.get_option('categorical_threshold')
-    numeric_categorical_threshold = ww.config.get_option('numeric_categorical_threshold')
+    if pdtypes.is_categorical_dtype(series.dtype):
+        return True
 
     if pdtypes.is_string_dtype(series.dtype) and not col_is_datetime(series):
-        # heuristics to predict this some other than categorical
         sample = get_inference_sample(series)
-        # catch cases where object dtype cannot be interpreted as a string
-        try:
-            avg_length = sample.str.len().mean()
-            if avg_length > categorical_threshold:
-                return False
-        except AttributeError:
-            pass
-        return True
+        categorical_threshold = ww.config.get_option('categorical_threshold')
+
+        return _is_categorical(sample, categorical_threshold)
+
+    if pdtypes.is_float_dtype(series.dtype) or pdtypes.is_integer_dtype(series.dtype):
+        numeric_categorical_threshold = ww.config.get_option('numeric_categorical_threshold')
+        if numeric_categorical_threshold is not None:
+            return _is_categorical(series, numeric_categorical_threshold)
+        else:
+            return False
```

Comment:
@tamargrey I think the overall way I was trying to structure things was so that the top-level if statements are trying to identify the overall category of Pandas dtype that a series might fall into (explicitly categorical?  some kind of object?  numerical?).  Then, within each of those if blocks, the function should figure out how to explicitly say whether or not it cares about the series and give a concrete answer.  From that, I think it's a *little* easier to follow the possible execution of the function in your head.  The reader of the function sees an if block that has a return statement in the positive case and also in the negative case and knows without any further examination that the function will return at that point.  Otherwise, they might have to read further down to figure out exactly what will happen.  We could save a line of code by not including the else block but I think we lose a little in readability, especially if the function gets updated later to include further logic before the final `return False` at the bottom of the function.

Reply:
Thanks for the explanation! That makes sense to me, happy to keep as is